### PR TITLE
Adds support for continuous integration on travis by ensuring that ev…

### PR DESCRIPTION
**Pull request by dbernstein**
…ery untagged commit to master  is deployed to the sonatype snapshot repository.

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3291

# How should this be tested?

Since this change involves updates to the .travis.yml file I believe the only way to be sure if it works is to merge it and see if the snapshot gets pushed to sonatype as expected.

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -76,6 +76,18 @@
     <!-- default properties that can be altered on the command line -->
     <fcrepo.test.context.path />
   </properties>
+
+  <distributionManagement>
```
Comment (awoods) [inline pom.xml:80]: I would these changes to live in: https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-parent/pom.xml

Comment (dbernstein) [reply pom.xml:80]: 👍

File: fcrepo-parent/pom.xml [hunk 2]
```diff
@@ -69,6 +69,17 @@
     <sonar.jacoco.itReportPath>${jacoco.outputDir}/${jacoco.out.it.file}</sonar.jacoco.itReportPath>
   </properties>
 
+  <distributionManagement>
```
Comment (awoods) [inline fcrepo-parent/pom.xml:72]: I would suggest using the existing `distributionManagement` block: https://github.com/fcrepo4/fcrepo4/blob/20520a066407778627cf8500b49d039fbd80599e/fcrepo-parent/pom.xml#L571-L586
..and potentially align the `serverId` (https://github.com/fcrepo4/fcrepo4/pull/1673/files#diff-44094b67d49071d47bb32efe43678722R402).

File: fcrepo-parent/pom.xml [hunk 3]
```diff
@@ -382,6 +393,33 @@
...
+          <configuration>
+            <serverId>ossrh</serverId>
+            <nexusUrl>https://oss.sonatype.org/</nexusUrl>
+            <autoReleaseAfterClose>true</autoReleaseAfterClose>
+          </configuration>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-gpg-plugin</artifactId>
```
Comment (awoods) [inline fcrepo-parent/pom.xml:410]: We already have this plugin: https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-parent/pom.xml#L497-L519

File: pom.xml [hunk 4]
```diff
@@ -76,6 +76,7 @@
     <!-- default properties that can be altered on the command line -->
     <fcrepo.test.context.path />
   </properties>
+
```
Comment (awoods) [inline pom.xml:79]: Can probably remove this file from this pull-request.

**Comment by awoods** [comment]
PR against this PR: https://github.com/dbernstein/fcrepo4/pull/9

**Pull request merged**